### PR TITLE
feat(React): remove component props that were deprecated in v10

### PR DESCRIPTION
**Pull request by jnm2377**
Rel #8418 

#### Changelog

- remove props that were deprecated as part of v10
- props that were deprecated in v11 (such as `light` among others) were **not** removed

#### Testing / Reviewing

- ensure all tests pass
- ensure components are working as expected:
    - NumberInput
    - OverflowMenuItem 
    - RadioTile
    - Search
    - Select
    - Slider
    - StructuredList
    - SelectableTile
    - TimePickerSelect

### CODE REVIEW

File: packages/react/src/components/NumberInput/next/NumberInput.js [hunk 1]
```diff
@@ -77,7 +75,6 @@ const NumberInput = React.forwardRef(function NumberInput(props, forwardRef) {
     [`${prefix}--number--light`]: light,
     [`${prefix}--number--nolabel`]: hideLabel,
     [`${prefix}--number--nosteppers`]: hideSteppers,
-    [`${prefix}--number--mobile`]: isMobile,
```
Comment (tay1orjones) [inline packages/react/src/components/NumberInput/next/NumberInput.js:80]: Did we get the styles deleted for this one too?

File: packages/react/src/components/Select/Select.js [hunk 2]
```diff
@@ -222,7 +212,13 @@ Select.propTypes = {
    * `true` to use the light version. For use on $ui-01 backgrounds only.
    * Don't use this to make tile background color same as container background color.
    */
-  light: PropTypes.bool,
+  light: FeatureFlags.enabled('enable-v11-release')
+    ? deprecate(
+        PropTypes.bool,
+        'The `light` prop for `Select` is no longer needed and has ' +
+          'been deprecated in v11. It will be moved in the next major release.'
```
Comment (tay1orjones) [inline packages/react/src/components/Select/Select.js]: I think the `light` prop warnings should include a link or instruction to use `Layer`. Maybe we link to somewhere in the migration guide?

**Review by tay1orjones — APPROVED** [review]
A couple comments. Are these all documented in the migration guide?

File: packages/react/src/components/NumberInput/next/NumberInput.js (hunk 1 above)
Comment (jnm2377) [reply packages/react/src/components/NumberInput/next/NumberInput.js:80]: @tay1orjones I checked the v11 styles and they had already been removed. 👍🏽

File: packages/react/src/components/Select/Select.js (hunk 2 above)
Comment (jnm2377) [reply packages/react/src/components/Select/Select.js]: I'll update! 👍🏽

File: packages/react/src/components/OverflowMenuItem/OverflowMenuItem.js [hunk 3]
```diff
@@ -195,7 +182,10 @@ export default class OverflowMenuItem extends React.Component {
               <TagToUse
                 {...other}
                 {...{
-                  'data-floating-menu-primary-focus': primaryFocus || null,
+                  'data-floating-menu-primary-focus':
+                    (FeatureFlags.enabled('enable-v11-release') &&
+                      other.primaryFocus) ||
+                    null,
```
Comment (joshblack) [inline packages/react/src/components/OverflowMenuItem/OverflowMenuItem.js]: With `primaryFocus` being removed, should this just be removed?

**Review by joshblack — APPROVED** [review]
Looks great! Just left one note, wasn't sure if it was relevant or not

**Review by dakahn — APPROVED** [review]

File: packages/react/src/components/OverflowMenuItem/OverflowMenuItem.js (hunk 3 above)
Comment (jnm2377) [reply packages/react/src/components/OverflowMenuItem/OverflowMenuItem.js]: @joshblack I wasn't sure if I should remove it completely bc there's no `/next` component for this, as in this file is used for v10 and v11, and it only needs to be removed for v11.

Comment (joshblack) [reply packages/react/src/components/OverflowMenuItem/OverflowMenuItem.js]: @jnm2377 if it helps, you can treat this file at the moment as "v11". So if you want to remove it for v11 go for it 👍  All the files for v10 are under the v10 branch now

**Pull request merged**